zio-http/src/main/scala/zhttp/http/Http.scala

```diff
@@ -574,11 +610,27 @@ object Http {
    */
   def fromOptionFunction[A]: PartialFromOptionFunction[A] = new PartialFromOptionFunction(())
 
+  /**
+   * Creates an HTTP that can serve files on the give path.
+   */
+  def fromPath(head: String, tail: String*): HttpApp[Any, Throwable] =
+    Http.fromFile(Paths.get(head, tail: _*).toFile)
+
   /**
    * Creates an Http app from a resource path
    */
-  def fromResource(path: String): HttpApp[Any, Throwable] =
-    Http.fromFile(new File(getClass.getResource(path).getPath))
+  def fromResource(path: String): HttpApp[Any, Throwable] = {
+    for {
+      path <- Http
+        .attempt(getClass.getResource(path) match {
```

Comment:
In my implementation (inspired by yours), `getClass.getResource(path)` was returning `null`, I had to use `getClass.getClassLoader.getResource(path)`